packages/core/src/components/forms/textArea.tsx

```diff
@@ -75,22 +79,29 @@ export class TextArea extends AbstractPureComponent2<TextAreaProps, ITextAreaSta
         this.props.inputRef,
     );
 
-    public componentDidMount() {
-        if (this.props.growVertically && this.textareaElement !== null) {
-            // HACKHACK: this should probably be done in getSnapshotBeforeUpdate
-            /* eslint-disable-next-line react/no-did-mount-set-state */
-            this.setState({
-                height: this.textareaElement?.scrollHeight,
-            });
+    private maybeSyncHeightToScrollHeight = () => {
+        if (this.props.growVertically && this.textareaElement != null) {
+            const { scrollHeight } = this.textareaElement;
+            if (scrollHeight > 0) {
+                this.setState({ height: scrollHeight });
+            }
         }
+    };
+
+    public componentDidMount() {
+        this.maybeSyncHeightToScrollHeight();
     }
 
-    public componentDidUpdate(prevProps: TextAreaProps) {
+    public componentDidUpdate(prevProps: TextAreaProps, _prevState: TextAreaState) {
         if (prevProps.inputRef !== this.props.inputRef) {
             setRef(prevProps.inputRef, null);
             this.handleRef = refHandler(this, "textareaElement", this.props.inputRef);
             setRef(this.props.inputRef, this.textareaElement);
         }
+
+        if (prevProps.value !== this.props.value || prevProps.style !== this.props.style) {
```

Comment:
note that `props.style` changes can cause layout changes which affect `scrollHeight`